drivers/include/tla2528.hpp

```diff
@@ -31,51 +31,86 @@ class tla2528
     digital_output_push_pull
   };
 
+  /**
+   * @param p_i2c_address i2c address configured on the tla, by default is set
+   * to the i2c address of no resistors attached to address config pins.
+   */
   tla2528(hal::i2c& p_i2c, hal::byte p_i2c_address = default_address);
+
   /**
    * @brief set what service a pin will provide
+   *
+   * @param p_channel which pin to configure, if out of range (>7) an exception
+   * will be thrown
+   *
+   * if adapters are made for a pin an exception may be thrown to prevent
+   * invalid behavior
    */
   void set_pin_mode(pin_mode p_mode, hal::byte p_channel);
 
-  // If channel is not set to correct config then undefined behavior will occur
   /**
-   * @brief set output level pin
+   * @brief set digital output level pin
+   *
+   * @param p_channel if out of range (>7) an exception will be thrown
```

Comment:
Which exception? Use the doxygen `@throws` command to explain which one. Works the same was at parameter.